### ISSUE TITLE
Завести чёрный список хешей... фотки в чёрный список

### ISSUE THREAD

**Issue by tebaly** [tebaly/freedomsex#63]
а вы разработчики давно могли бы сделать сохранение фоток по хешу и завести чёрный список хешей... фотки в чёрный список добавляли бы администраторы в ручную с учётом всех обстоятельств... если человек отправляет запрещенную фоту, автоматом в бан...

**Comment by tebaly** [comment]
На время разработки такого функционала и для охлаждения фото-спамеров 
* активирован "усложняющий" фильтр при отправке фотографий

Он был спроектирован заранее, но потом, в виду отсутствия явного злоупотребления, был деактивирован. Протестируйте отправку фотографий пожалуйста при желании. Отправить их "малознакомому" стало сложнее. Со "знакомыми" осталось без ограничений. Возможно дальнейшее усложнение
* Фото отправляется сразу - единственное сообщение-фотография (простая капча)
* Фото отправлено "малознакомому" Пользователю (простая капча)

Можно просить ещё Гуглокапчу

**Comment by inquisitor6789** [comment]
на счет хэша для фоток у меня что-то большие сомнения, что это будет адекватно работать в глобальном масштабе. Для саши минского это сработает (он шлет одно и то же). А как быть с проститутками, которые один и тот же текст накладывают на разные фото?
Гугл и то не все ищет в отличие от яндекса. Или мы замахнемся и переплюнем гугл? :-)

**Comment by tebaly** [comment]
> Или мы замахнемся и переплюнем гугл? :-)

А вы уже где-то встречали аналогичный нашему проект по части администрирования?) Вот вы смеетесь, а мы то с вами на голову выше всех остальных.

> на счет хэша для фоток у меня что-то большие сомнения

Так особо и не заморачивались - сделать крайне сложно и ресурсоемко, а пользы... Но можно. Хотя бы откровенные дубли побанит.

**Comment by inquisitor6789** [comment]
> и какой такой непоправимый вред он может нанести?

нет. Да и если честно, все сайты знакомств которые я встречал, обходятся практически без администрирования (в том понимании как это есть на freedomsex). Но минус этого - они заполнены проституками. Или почти весь функционал платный.

**Comment by mj4444ru** [comment]
сложно делать нет смысла, всё равно обойдут если захотят, надо что то примитивное типа хеша... 
в отношении сообщений это уже у вас реализовано, так что тут понятно что и как...

**Comment by tebaly** [comment]
контрольная сумма файла должна подойти

**Comment by tebaly** [comment]
я бы сделал так:

1. при поступлении 5 жалоб на фото, фото автоматически блокируется (желательно по хешу) и скрывается у всех пользователей, при этом хозяин ничего не должен заподозрить и должен думать что он и дальше успешно шлёт фото...
2. если администратор видит такое фото, то он его блокирует (желательно по хешу) и оно скрывается у всех пользователей, при этом хозяин ничего не должен заподозрить и должен думать что он и дальше успешно шлёт фото...
